RecoPixelVertexing/PixelVertexFinding/plugins/gpuFitVertices.h

```diff
@@ -12,28 +12,32 @@
 
 namespace gpuVertexFinder {
 
-  __device__ __forceinline__ void fitVertices(ZVertices* pdata,
-                                              WorkSpace* pws,
+  __device__ __forceinline__ void fitVertices(VtxSoAView pdata,
+                                              WsSoAView pws,
                                               float chi2Max  // for outlier rejection
   ) {
     constexpr bool verbose = false;  // in principle the compiler should optmize out if false
 
-    auto& __restrict__ data = *pdata;
-    auto& __restrict__ ws = *pws;
-    auto nt = ws.ntrks;
-    float const* __restrict__ zt = ws.zt;
-    float const* __restrict__ ezt2 = ws.ezt2;
-    float* __restrict__ zv = data.zv;
-    float* __restrict__ wv = data.wv;
-    float* __restrict__ chi2 = data.chi2;
-    uint32_t& nvFinal = data.nvFinal;
-    uint32_t& nvIntermediate = ws.nvIntermediate;
-
-    int32_t* __restrict__ nn = data.ndof;
-    int32_t* __restrict__ iv = ws.iv;
-
-    assert(pdata);
+    auto& __restrict__ data = pdata;
+    auto& __restrict__ ws = pws;
+    auto nt = ws.ntrks();
+    float const* __restrict__ zt = ws.zt();
+    float const* __restrict__ ezt2 = ws.ezt2();
+    float* __restrict__ zv = data.zv();
+    float* __restrict__ wv = data.wv();
+    float* __restrict__ chi2 = data.chi2();
```

Comment:
here and elsewhere: I _think_ we can avoid doing the many variations of
```c++
type* __restrict__ x = view.x();
```
and use `view.x()` directly in the code.

@ericcano's SoA View should already mark the pointers to the data as `__restrict__` in order to generate optimal code.


However, I'm fine making this specific change in a separate PR, which would also make it easy to check its impact (none expected) on the generated code and performance.